### ISSUE TITLE
Wrong window scale on KDE Plasma

### ISSUE THREAD

**Issue by Jacalz** [fyne-io/fyne#4886]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

The calculations for determining the default Window scale seem to not be working on KDE Plasma. This was also reported as a Steam Deck specific issue over at https://github.com/Jacalz/rymdport/issues/159 but I can replicate it in a VM on my Gnome desktop also. On a small screen like the Steam Deck, this is as big of a problem that it makes the entire application unusable.

I have attached various comparison images below to showcase this issue.

### How to reproduce

1. Start fyne_demo and notice that the scaling is incorrect.
2. Open the theme settings and change scaling.

### Screenshots

As you can see, the scale is entirely different from Gnome and it isn't quite possible to get it to a good scale factor when using the settings. The comparison images used Rymdport but fyne_demo should work fine as well. 

## Gnome vs Plasma window with normal text size:
![image](https://github.com/fyne-io/fyne/assets/25466657/2026b7b3-9307-49a1-8e12-58ff1617b931)

## Setting text size to small on Plasma is still too large:
![image](https://github.com/fyne-io/fyne/assets/25466657/52faa8d2-8ebf-46fe-ae63-4e593efec97f)

## Setting text size to tiny on Plasma is too small:
![image](https://github.com/fyne-io/fyne/assets/25466657/9476b47a-23ec-42d2-9fc0-2be1d1e654b6)

### Example code

Use fyne_demo and the menu to then open the theme settings.

### Fyne version

v2.4.5

### Go compiler version

1.22.3

### Operating system and version

Fedora Kinoite 40

## Display hardware

All of these tests were conducted on a 27" 1440p monitor.

## Reported scaling values on Gnome:

![image](https://github.com/fyne-io/fyne/assets/25466657/f1823200-804f-4b7b-a422-7abe8634cffa)

## Reported scaling values on Plasma:

![image](https://github.com/fyne-io/fyne/assets/25466657/2866b95c-24a0-451d-9cd6-31e86b07671c)

**Comment by Jacalz** [comment]
This looks really bad. Given how unusable this makes Fyne apps on Plasma, I'd almost be tempted to label this as a release blocker.

**Comment by andydotxyz** [comment]
> Gnome vs Plasma window with normal text size:

How are you managing to run both on the same screen?

Is it possible that one is X11 and the other Wayland?

**Comment by dweymouth** [comment]
I agree with Jacob - this is really bad and should be one of the top priority fixes. It does seem like Fyne apps are almost entirely unusable on KDE Plasma as it is now

**Comment by dweymouth** [comment]
I'm surprised I haven't had anyone report this though for Supersonic

**Comment by Jacalz** [comment]
> In which case I'd guess this is a duplicate of the issue that Wayland scale is not yet respected.

You mean https://github.com/fyne-io/fyne/issues/2897? Not necessarily the same given that KDE and Gnome gives entirely different scaling factors. That the application is unusable on KDE is really problematic.

> For scale lookup we use the XRandR information so if this is running as XWayland instead of Wayland then I think that may be the source of the problem. Would be worth getting a report on the xRandR screen information to aid debugging...

Neither Gnome or Plasma installs have xrandr installed so I wonder where we get our scaling info from in that case. I installed it and I have attached the reports below.

Xrandr report from Gnome:
```
Screen 0: minimum 16 x 16, current 4480 x 1440, maximum 32767 x 32767
DP-2 connected primary 2560x1440+1920+0 (normal left inverted right x axis y axis) 600mm x 340mm
```

Xrandr report from KDE Plasma:
```
Screen 0: minimum 16 x 16, current 2560 x 1440, maximum 32767 x 32767
DP-2 connected primary 2560x1440+0+0 (normal left inverted right x axis y axis) 320mm x 200mm
```

**Comment by Jacalz** [comment]
Opened https://github.com/fyne-io/fyne/issues/4896 to track Steam Deck changes instead. Relabeled this issue for incorrect scaling in virtual machines.

**Comment by andydotxyz** [comment]
It works for my testing in full virtual machines. What does the Gnome boxes provide by way of monitor / hardware emulation - we read the info from xrandr so I'm not sure where it's likely to be falling over.

**Comment by Jacalz** [comment]
The virtual machines that Gnome Boxes manages are full virtual machine as far as I know. It is basically a virtualisation manager for libvirt. I believe it is running either through QEMU or KVM. See https://en.m.wikipedia.org/wiki/GNOME_Boxes for more information.

I suppose you could try to install Gnome Boxes on a Linux machine and see if it replicates for you.

**Comment by andydotxyz** [comment]
We probably need to know more about XWayland, because it is correctly reading the scale info in a pure Wayland environment...
